content/guides/javascript/get-program-accounts.md

```diff
@@ -0,0 +1,307 @@
+    [State commitment](https://docs.solana.com/developing/clients/jsonrpc-api#configuring-state-commitment)
+  - (optional) `encoding`: `string` - Encoding for account data, either:
+    `base58`, `base64`, or `jsonParsed`. Note, web3js users should instead use
+    [getParsedProgramAccounts](https://solana-labs.github.io/solana-web3.js/classes/Connection.html#getParsedProgramAccounts)
+  - (optional) `dataSlice`: `object` - Limit the returned account data based on:
+    - `offset`: `number` - Number of bytes into account data to begin returning
+    - `length`: `number` - Number of bytes of account data to return
+  - (optional) `filters`: `array` - Filter results using the following filter
+    objects:
+    - `memcmp`: `object` - Match a series of bytes to account data:
+      - `offset`: `number` - Number of bytes into account data to begin
+        comparing
+      - `bytes`: `string` - Data to match, as base58 encoded string limited to
+        129 bytes
+    - `dataSize`: `number` - Compares the account data length with the provided
+      data size
+  - (optional) `withContext`: `boolean` - Wrap the result in an
+    [RpcResponse JSON object](https://docs.solana.com/developing/clients/jsonrpc-api#rpcresponse-structure)
+```
+
+### Response
+
+By default `getProgramAccounts` will return an array of JSON objects with the
+following structure:
+
+- `pubkey`: `string` - The account pubkey as a base58 encoded string
+- `account`: `object` - a JSON object, with the following sub fields:
+  - `lamports`: `number`, number of lamports assigned to the account
+  - `owner`: `string`, The base58 encoded pubkey of the program the account has
+    been assigned to
+  - `data`: `string` | `object` - data associated with the account, either as
+    encoded binary data or JSON format depending on the provided encoding
+    parameter
+  - `executable`: `boolean`, Indication if the account contains a program
+  - `rentEpoch`: `number`, The epoch at which this account will next owe rent
+
+## Deep Dive
+
+`getProgramAccounts` is a versatile RPC method that returns all accounts owned
+by a program. We can use `getProgramAccounts` for a number of useful queries,
+such as finding:
+
+- All token accounts for a particular wallet
+- All token accounts for a particular mint (i.e. All
+  [SRM](https://www.projectserum.com/) holders)
+- All custom accounts for a particular program (i.e. All
+  [Mango](https://mango.markets/) users)
+
+Despite its usefulness, `getProgramAccounts` is often misunderstood due to its
+current constraints. Many of the queries supported by `getProgramAccounts`
+require RPC nodes to scan large sets of data. These scans are both memory and
+resource intensive. As a result, calls that are too frequent or too large in
+scope can result in connection timeouts. Furthermore, at the time of this
+writing, the `getProgramAccounts` endpoint does not support pagination. If the
+results of a query are too large, the response will be truncated.
```

Comment:
Worth double checking (since this content mentions Serum and that's 2 years ago) - is there still no pagination?

Reply:
I don't think so.
https://github.com/solana-labs/solana/issues/19187

Reply:
I would love to know though, searched for a lot of pages.